### PR TITLE
CAT-2356 Automatically logout non-NOLB users

### PR DESCRIPTION
**Pull request by sjaindl**

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java [hunk 1]
```diff
@@ -390,6 +390,12 @@ public void handleUploadButton(View view) {
 		if (!viewSwitchLock.tryLock()) {
 			return;
 		}
+
+		if (BuildConfig.CREATE_AT_SCHOOL && !Utils.isCreateAtSchoolUser(this)) {
+			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+			sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).apply();
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java]: (I know that this is already like this in other parts of the code)

if you edit multiple items you should call in this order:
```
SharedPreferences.Editor prefEditor = sharedPreferences.edit();
prefEditor.putString();
prefEditor.putString();
//... all other operations on this editor
prefEditor.apply();
```
for the following reasons: 

- apply() is an atomic write operation, meaning no part of the app could access the preferences in a weird state where for example Constants.TOKEN is written, but Constants.USERNAME still has the old value.
- Each time you call edit() a new SharedPreferences.Editor is created which creates a full copy of the original preferences and each apply() writes your modiefied copy back to the original -> performance

please check all your code bits which use this

File: catroid/src/androidTest/java/org/catrobat/catroid/test/utils/IconsUtilTest.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ public void testAddIcons() {
 		assertNotNull("Control pos drawable not found.", correctDrawable);
 
 		Bitmap bitmap = ((BitmapDrawable) drawables[0]).getBitmap();
-		assertEquals("Icon size did not match the expected size.", IconsUtil.smallIconSizeBricks, drawables[0].getBounds());
+		assertEquals("Icon size did not match the expected size.", IconsUtil.getSmallIconSizeBricks(), drawables[0].getBounds());
```
Comment (cheidenreich) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/utils/IconsUtilTest.java]: why is something icon related done in this PR? 😛

File: catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java [hunk 3]
```diff
@@ -817,6 +817,16 @@ public boolean login(String username, String password, String token, Context con
 			String serverAnswer = jsonObject.optString(JSON_ANSWER);
 			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
 
+			if (BuildConfig.CREATE_AT_SCHOOL) {
+				boolean createAtSchoolUser = jsonObject.optBoolean(Constants.CREATE_AT_SCHOOL_USER);
+				sharedPreferences.edit().putBoolean(Constants.CREATE_AT_SCHOOL_USER, createAtSchoolUser).apply();
+				if (!createAtSchoolUser) {
+					sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).apply();
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java]: is there a reason not to call `Utils.logoutUser()`?

Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java]: @sjaindl please check that, after this we can merge, if your next PR is the bugfix for invalid entries ;)

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java (hunk 1 above)
Comment (sjaindl) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java]: That's a really good hint! I've changed it accordingly

File: catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java (hunk 3 above)
Comment (sjaindl) [inline catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java]: Yes, now I remember the reason --> In the create@school-flavor there is an additional Tracking done in Utils.logoutUser() -- TrackingUtil.trackLogoutEndSessionEvent(context);
This should not be logged in the case of an error here.
Facebook and Google+ logout isn't relevant, as only native login is provided in this flavor. Furthermore, we don't want to show the Toast message "Logout successful". So it should be perfectly fine here to reset those 2 entries in the shared prefs.

File: catroid/src/androidTest/java/org/catrobat/catroid/test/utils/IconsUtilTest.java (hunk 2 above)
Comment (sjaindl) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/utils/IconsUtilTest.java]: Well, that was just a fix for not compiling tests. But yeah you're right, normally this should be a separate PR :P

File: catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java (hunk 3 above)
Comment (sjaindl) [inline catroid/src/main/java/org/catrobat/catroid/web/ServerCalls.java]: Yes, next I'll have a look at the invalid entries bug ;)
